cgi-bin/DW/Controller/Customize.pm

```diff
@@ -0,0 +1,654 @@
+#!/usr/bin/perl
+#
+# DW::Controller::Customize
+#
+# This controller is for customize handlers.
+#
+# Authors:
+#      R Hatch <ruth.s.hatch@gmail.com>
+#
+# Copyright (c) 2016 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself. For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+package DW::Controller::Customize;
+
+use strict;
+use warnings;
+use DW::Controller;
+use DW::Routing;
+use DW::Template;
+use DW::Logic::MenuNav;
+use JSON;
+use Carp;
+
+DW::Routing->register_string( '/customize/',        \&customize_handler, app => 1 );
+DW::Routing->register_string( '/customize/options', \&options_handler,   app => 1 );
+
+DW::Routing->register_rpc( "themechooser",  \&themechooser_handler,  format => 'json' );
+DW::Routing->register_rpc( "journaltitles", \&journaltitles_handler, format => 'html' );
+DW::Routing->register_rpc( "layoutchooser", \&layoutchooser_handler, format => 'json' );
+DW::Routing->register_rpc( "themefilter",   \&filter_handler,        format => 'json' );
+
+sub customize_handler {
+    my ( $ok, $rv ) = controller( authas => 1, form_auth => 1 );
+    return $rv unless $ok;
+
+    my $r      = $rv->{r};
+    my $post   = $r->post_args;
+    my $u      = $rv->{u};
+    my $remote = $rv->{remote};
+    my $GET    = $r->get_args;
+
+    my $vars;
+    $vars->{u}            = $u;
+    $vars->{remote}       = $remote;
+    $vars->{is_identity}  = 1 if $u->is_identity;
+    $vars->{is_community} = 1 if $u->is_community;
+    $vars->{style}        = LJ::Customize->verify_and_load_style($u);
+    $vars->{authas_html}  = $rv->{authas_html};
+
+    my $queryargs = make_queryargs($GET);
+
+    if ( $u ne $remote ) {
+        $queryargs->{authas} = $u->user;
+    }
+
+    # lazy migration of style name
+    LJ::Customize->migrate_current_style($u);
+
+    # set up the keywords for basic search
+    my @keywords = LJ::Customize->get_search_keywords_for_js($u);
+    $vars->{autocomplete} = \@keywords;
+
+    # we want to have "All" selected if we're filtering by layout or designer, or if we're searching
+    $vars->{viewing_all} = $queryargs->{layoutid} || $queryargs->{designer} || $queryargs->{search};
+
+    # sort cats by specificed order key, then alphabetical order
+    my %cats = LJ::Customize->get_cats($remote);
```

Comment:
Is this intentionally `$remote` here? Since we're operating on `$u` otherwise.

Reply:
Oh, good catch, checking the old code it looks like should probably be $u (though uh in our case it won't make any difference, all users get the same theme cats)